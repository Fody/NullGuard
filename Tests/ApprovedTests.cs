#if(DEBUG)

using System.Linq;
using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using NUnit.Framework;

[TestFixture]
[UseReporter(typeof(DiffReporter), typeof(ClipboardReporter))]
[UseApprovalSubdirectory("approvals")]
public class ApprovedTests
{
    [Test]
    public void ClassWithBadAttributes()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPaths[0], "ClassWithBadAttributes"));
    }

    [Test]
    public void ClassWithPrivateMethod()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPaths[0], "ClassWithPrivateMethod"));
    }

    [Test]
    public void ClassWithPrivateMethodNoAssert()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPaths[1], "ClassWithPrivateMethod"));
    }

    [Test]
    public void GenericClass()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPaths[0], "GenericClass`1"));
    }

    [Test]
    public void GenericClassNoAssert()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPaths[1], "GenericClass`1"));
    }

    [Test]
    public void Indexers()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPaths[0], "Indexers"));
    }

    [Test]
    public void InterfaceBadAttributes()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPaths[0], "InterfaceBadAttributes"));
    }

    [Test]
    public void SimpleClass()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPaths[0], "SimpleClass"));
    }

    [Test]
    public void SimpleClassNoAssert()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPaths[1], "SimpleClass"));
    }

    [Test]
    public void SkipIXamlMetadataProvider()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPaths[0], "XamlMetadataProvider"));
    }

    [Test]
    public void SpecialClass()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPaths[0], "SpecialClass"));
    }

    [Test]
    public void SpecialClassNoAssert()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPaths[1], "SpecialClass"));
    }

    [Test]
    public void PublicNestedInsideNonPublic()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPaths[0], "NonPublicWithNested"));
    }

    [Test]
    public void UnsafeClass()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPaths[0], "UnsafeClass"));
    }

    [Test]
    public void SampleClassWithDisabledInjectJetBrainsAnnotations()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPaths[0], "SampleClassWithDisabledInjectJetBrainsAnnotations"));
    }

    [Test]
    public void SampleClassWithoutJetBrainsAnnotationsAttribute()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPaths[3], "SampleClassWithoutJetBrainsAnnotationsAttribute"));
    }

    [Test]
    public void InfosList()
    {
        Approvals.VerifyAll(AssemblyWeaver.Infos.OrderBy(e => e), "Infos: ");
    }

    [Test]
    public void WarnsList()
    {
        Approvals.VerifyAll(AssemblyWeaver.Warns.OrderBy(e => e), "Warns: ");
    }

    [Test]
    public void ErrorsList()
    {
        Approvals.VerifyAll(AssemblyWeaver.Errors.OrderBy(e => e), "Errors: ");
    }
}

#endif