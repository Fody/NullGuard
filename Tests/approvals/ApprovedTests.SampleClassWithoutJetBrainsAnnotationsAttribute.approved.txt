.class public auto ansi beforefieldinit SampleClassWithoutJetBrainsAnnotationsAttribute
       extends [mscorlib]System.Object
{
  .field private string '<SomeProperty>k__BackingField'
  .custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
  .method public hidebysig instance string 
          SomeMethod(string nonNullArg) cil managed
  {
    // Code size       70 (0x46)
    .maxstack  3
    .language '{3F5162F8-07C6-11D3-9053-00C04FA302A1}', '{994B45C4-E6E9-11D2-903F-00C04FA302A1}', '{5A869D0B-6611-11D3-BD2A-0000F80849BD}'
    IL_0000:  ldarg.1
    IL_0001:  ldnull
    IL_0002:  ceq
    IL_0004:  ldc.i4.0
    IL_0005:  ceq
    IL_0007:  ldstr      "[NullGuard] nonNullArg is null."
    IL_000c:  call       void [System]System.Diagnostics.Debug::Assert(bool,
                                                                       string)
    IL_0011:  ldarg.1
    IL_0012:  brtrue.s   IL_0024
    IL_0014:  ldstr      "nonNullArg"
    IL_0019:  ldstr      "[NullGuard] nonNullArg is null."
    IL_001e:  newobj     instance void [mscorlib]System.ArgumentNullException::.ctor(string,
                                                                                     string)
    IL_0023:  throw
    IL_0024:  ldnull
    IL_0025:  dup
    IL_0026:  ldnull
    IL_0027:  ceq
    IL_0029:  ldc.i4.0
    IL_002a:  ceq
    IL_002c:  ldstr      "[NullGuard] Return value of method 'System.String "
    + "SampleClassWithoutJetBrainsAnnotationsAttribute::SomeMethod(System.Stri"
    + "ng)' is null."
    IL_0031:  call       void [System]System.Diagnostics.Debug::Assert(bool,
                                                                       string)
    IL_0036:  dup
    IL_0037:  brtrue.s   IL_0045
    IL_0039:  pop
    IL_003a:  ldstr      "[NullGuard] Return value of method 'System.String "
    + "SampleClassWithoutJetBrainsAnnotationsAttribute::SomeMethod(System.Stri"
    + "ng)' is null."
    IL_003f:  newobj     instance void [mscorlib]System.InvalidOperationException::.ctor(string)
    IL_0044:  throw
    IL_0045:  ret
  } 
  .method public hidebysig specialname instance string 
          get_SomeProperty() cil managed
  {
    .custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
    // Code size       39 (0x27)
    .maxstack  3
    IL_0000:  ldarg.0
    IL_0001:  ldfld      string SampleClassWithoutJetBrainsAnnotationsAttribute::'<SomeProperty>k__BackingField'
    IL_0006:  dup
    IL_0007:  ldnull
    IL_0008:  ceq
    IL_000a:  ldc.i4.0
    IL_000b:  ceq
    IL_000d:  ldstr      "[NullGuard] Return value of property 'System.Strin"
    + "g SampleClassWithoutJetBrainsAnnotationsAttribute::SomeProperty()' is n"
    + "ull."
    IL_0012:  call       void [System]System.Diagnostics.Debug::Assert(bool,
                                                                       string)
    IL_0017:  dup
    IL_0018:  brtrue.s   IL_0026
    IL_001a:  pop
    IL_001b:  ldstr      "[NullGuard] Return value of property 'System.Strin"
    + "g SampleClassWithoutJetBrainsAnnotationsAttribute::SomeProperty()' is n"
    + "ull."
    IL_0020:  newobj     instance void [mscorlib]System.InvalidOperationException::.ctor(string)
    IL_0025:  throw
    IL_0026:  ret
  } 
  .method public hidebysig specialname instance void 
          set_SomeProperty(string 'value') cil managed
  {
    .custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
    // Code size       44 (0x2c)
    .maxstack  2
    IL_0000:  ldarg.1
    IL_0001:  ldnull
    IL_0002:  ceq
    IL_0004:  ldc.i4.0
    IL_0005:  ceq
    IL_0007:  ldstr      "[NullGuard] Cannot set the value of property 'Syst"
    + "em.String SampleClassWithoutJetBrainsAnnotationsAttribute::SomeProperty"
    + "()' to null."
    IL_000c:  call       void [System]System.Diagnostics.Debug::Assert(bool,
                                                                       string)
    IL_0011:  ldarg.1
    IL_0012:  brtrue.s   IL_0024
    IL_0014:  ldstr      "value"
    IL_0019:  ldstr      "[NullGuard] Cannot set the value of property 'Syst"
    + "em.String SampleClassWithoutJetBrainsAnnotationsAttribute::SomeProperty"
    + "()' to null."
    IL_001e:  newobj     instance void [mscorlib]System.ArgumentNullException::.ctor(string,
                                                                                     string)
    IL_0023:  throw
    IL_0024:  ldarg.0
    IL_0025:  ldarg.1
    IL_0026:  stfld      string SampleClassWithoutJetBrainsAnnotationsAttribute::'<SomeProperty>k__BackingField'
    IL_002b:  ret
  } 
  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       7 (0x7)
    .maxstack  1
    IL_0000:  ldarg.0
    IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
    IL_0006:  ret
  } 
  .property instance string SomeProperty()
  {
  } 
}
